src/pages/Account/Tabs/ModulesTab.tsx

```diff
@@ -179,6 +198,230 @@ function ViewCode({address}: {address: string}): JSX.Element {
+                      ? grey[500]
+                      : grey[200],
+                    ":hover": {
+                      cursor: "pointer",
+                    },
+                  }}
+                >
+                  {fn.name}
+                </Box>
+              );
+            })}
+          </Box>
+        ))}
+      </Box>
+    </Box>
+  );
+}
+
+function WriteContractForm({
+  module,
+  fn,
+}: {
+  module: Types.MoveModule;
+  fn: Types.MoveFunction;
+}) {
+  const [state] = useGlobalState();
+  const {account, connected} = useWallet();
+  const {handleSubmit, control} = useForm<WriteContractFormType>();
+  const {submitTransaction} = useSubmitTransaction();
+
+  const onSubmit: SubmitHandler<WriteContractFormType> = async (data) => {
+    const payload: Types.TransactionPayload = {
+      type: "entry_function_payload",
+      function: `${module.address}::${module.name}::${fn.name}`,
+      type_arguments: data.typeArgs,
+      arguments: data.args,
+    };
+    await submitTransaction(payload);
+  };
+
+  const hasSigner = fn.params.length > 0 && fn.params[0] === "&signer";
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Stack spacing={4}>
+        <Typography fontSize={14}>
+          {fn.name}
+          {fn.generic_type_params.length > 0 &&
+            "<" +
+              [...Array(fn.generic_type_params.length)].map((_, i) => `T${i}`) +
+              ">"}
+          ({fn.params.join(", ")})
+        </Typography>
+        <Stack spacing={4}>
+          {[...Array(fn.generic_type_params.length)].map((_, i) => (
+            <Controller
+              key={i}
+              name={`typeArgs.${i}`}
+              control={control}
+              render={({field: {onChange, value}}) => (
+                <TextField
+                  onChange={onChange}
+                  value={value}
+                  label={`T${i}`}
+                  fullWidth
+                />
+              )}
+            />
+          ))}
+          {fn.params.map((param, i) => {
+            if (i === 0 && hasSigner) {
+              return (
+                <TextField
+                  key={i}
+                  fullWidth
+                  disabled
+                  value={account?.address ?? ""}
+                  label={"account: &signer"}
+                />
+              );
+            }
+            return (
+              <Controller
+                key={i}
+                name={`args.${hasSigner ? i - 1 : i}`}
+                control={control}
+                render={({field: {onChange, value}}) => (
+                  <TextField
+                    onChange={onChange}
+                    value={value}
+                    label={`arg${hasSigner ? i - 1 : i}: ${param}`}
+                    fullWidth
+                  />
+                )}
+              />
+            );
+          })}
+        </Stack>
+        {connected ? (
+          <Button type="submit" variant="contained">
```

Comment:
<img width="902" alt="Screenshot 2023-03-09 at 3 39 20 PM" src="https://user-images.githubusercontent.com/121921928/224185076-944bc537-a18e-40a0-a800-af6c829162ce.png">
lets fix the button width?